sky/tools/skyanalyzer

```diff
@@ -32,6 +32,9 @@ _IGNORED_PATTERNS = [
   re.compile(r'^\[hint\] .+ requires dynamic invoke'), # too many false positives, e.g. https://github.com/dart-lang/sdk/issues/24564
   re.compile(r'^\[hint\] Runtime check on non-ground type .+ may throw StrongModeError'), # https://github.com/dart-lang/sdk/issues/24565
   re.compile(r'^\[hint] \([_, ]+\) .+ has inferred type '), # ignore underscore-only arguments
+  re.compile(r'^\[hint] .+ has inferred type \(.+\) '), # ignore inferred function types
```

Comment:
The problem is if you don't type the list literal, the constructed object has the wrong type:
```dart
void main() {
  List a;
  a = [1, 2, 3];
  print(a.runtimeType);
  a = <int>[1, 2, 3];
  print(a.runtimeType);
}
```

This means if later you compare the type, you might get the wrong result:
```dart
void main() {
  List a;
  a = [1, 2, 3];
  print(a is List<int>); // true                                                                                                                                                                                           
  print(a is List<String>); // true (!)                                                                                                                                                                                    
  a = <int>[1, 2, 3];
  print(a.runtimeType);
  print(a is List<int>); // true                                                                                                                                                                                           
  print(a is List<String>); // false                                                                                                                                                                                       
}
```

In other words, despite the claims of the hint, the type isn't really inferred. Consider:

```dart
void main() {
  List a;
  a = [1, 2, 3];
  a.add('hello'); // "works" fine, no warnings or errors
  a = <int>[1, 2, 3];
  a.add('hello'); // analyzer hint, checked mode exception                                                                                                                                                                 
}
```